src/session/client-certificate.c

```diff
@@ -300,8 +368,39 @@ cockpit_session_client_certificate_map_user (const char *client_certificate_file
       return NULL;
     }
 
+  /* check if stdin is a Unix socket; then we are being called via cockpit-session@.service and need to
+   * check the cgroup of our peer */
+  struct ucred ucred;
+  int ws_pid_dirfd;
+  if (getsockopt (STDIN_FILENO, SOL_SOCKET, SO_PEERCRED, &ucred, &(socklen_t){sizeof ucred}) == 0)
+    {
+      debug ("unix socket mode, reading cgroup from peer pid %d", ucred.pid);
+      ws_pid_dirfd = open_proc_pid (ucred.pid);
+      int my_pid_dirfd = open_proc_pid (getpid ());
```

Comment:
Minor style: I'd find it a bit easier to read if we opened our pid, read our start time, and closed our pid in a vertical-whitespace-separated block, before moving on to deal with the ws pid stuff.  Maybe even before the `getsockopt()` call (see next comment below).

Reply:
Sure! I rearranged this a bit. It's not quite as simple just yet as you hope in *this* commit, but I can simplify it further when dropping support for cert auth for non-unix-socket mode [above](https://github.com/cockpit-project/cockpit/pull/16808#discussion_r1849911458).